substrate/client/rpc-spec-v2/src/chain_head/chain_head_follow.rs

```diff
@@ -702,7 +707,10 @@ where
 		let merged = tokio_stream::StreamExt::merge(merged, stream_responses);
 		let stream = stream::once(futures::future::ready(initial)).chain(merged);
 
-		self.submit_events(&startup_point, stream.boxed(), pruned_forks, sink, sub_data.rx_stop)
-			.await
+		// These are the pruned blocks that we should not report again.
+		for pruned in pruned_forks {
```

Comment:
Just to be sure I understand this correctly. When we generate the initial events we consider all blocks that are part of a fork that started below the finalized one as pruned, even if they are technically not pruned yet. So these will never be reported?

Reply:
Yep. I can't remember exactly, this was introduced quite a while ago to handle the case where the `Finalized` event would contain a `pruned block` not reported by a new block event. 

This abuses the purpose of `self.pruned_blocks`, which now holds:
- (previously stored in to_ignore)  forks that we did not report previously by the Initialized event
- (the intended purpose of this PR) pruned blocks already reported once